docs/4_2_co-simulation_api.adoc

```diff
@@ -1,387 +1,134 @@
+Compared to FMI for Model Exchange, <<fmi3ExitInitializationMode>> enters directly into <<StepMode>>.
+This was designed in case the Co-Simulation FMU does not support <<EventMode>>.
+Importers can directly switch a Co-Simulation FMU into <<EventMode>> without calling <<fmi3DoStep>> to handle events immediately after initialization.
 
-In order to retrieve the status of <<outputClock,`output clocks`>>, <<fmi3GetClock>> and <<fmi3GetIntervalDecimal>> or <<fmi3GetIntervalFraction>> need to be called in the *Event Mode*.
-If the <<fmi3DoStep>> return value is <<fmi3OK>>, the calling of <<fmi3GetClock>>, <<fmi3GetIntervalDecimal>>, <<fmi3GetIntervalFraction>>, <<fmi3UpdateDiscreteStates>> is only meaningful after <<fmi3SetClock>> in the case of super-dense time iterations are desired.
+==== State: Step Mode [[StepMode,*Step Mode*]]
 
-Similar to the Model Exchange case, the allowed call order is <<fmi3GetClock>>, <<fmi3GetIntervalDecimal>>, <<fmi3GetIntervalFraction>>, `fmi3Get{VariableType}`, `fmi3Set{VariableType}`.
-Function calls of this call order can be omitted.
+The state <<StepMode>> is used to compute the values of all variables between communication points and events and to advance time in the FMU.
+If the FMU is connected in loops with other models, iterations over the FMU equations requires setting the FMU to a previous communication point and repeating the <<fmi3DoStep>>.
 
-The handling of return values of function calls is identical to Co-Simulation.
+[cols="2,1",options="header",]
+|====
+|Equations and Actions
+|Functions Influencing Equations
 
-If <<terminateSimulation,`terminateSimulation`>> becomes `fmi3True` after calling <<fmi3UpdateDiscreteStates>> then the co-simulation should be terminated by calling <<fmi3Terminate>>.
-Once handling of the <<clock>> events finished, the co-simulation algorithm calls <<fmi3EnterStepMode>> for that FMU to push it into *Step Mode*.
-Note that it is not allowed to call <<fmi3EnterEventMode>> or <<fmi3EnterStepMode>> in Scheduled Execution.
+|Set <<tunable>> <<parameter,`parameters`>> latexmath:[\mathbf{p}_{\mathit{variability == tunable}}] .
+|`fmi3Set{VariableType}`
 
-_[Usually the co-simulation algorithm should be able to derive (but is not forced to do so) the correct communication point times for <<inputClock,`input clocks`>> in advance and thus it should be able to set the proper <<communicationStepSize>> for <<fmi3DoStep>>._
-_This might not be possible if an <<periodic,aperiodic>> <<inputClock>> of an FMU depends on the ticking of an <<periodic,aperiodic>> <<outputClock>> of another FMU or other <<periodic,aperiodic>> tick sources.]_
+|Set continuous-time and discrete-time <<input,`inputs`>> latexmath:[\mathbf{u}_{c+d}(\mathbf{t})].
+|`fmi3Set{VariableType}`
 
-==== State Machine for Co-Simulation [[state-machine-co-simulation]]
+|Get values of variables latexmath:[\mathbf{v}(\mathbf{t})].
+|`fmi3Get{VariableType}`
```

Comment:
This violates our calling sequence of fmi3SetXX fmi3DoStep fmi3GetXX, order should be changed!

Reply:
See above: no order intended. Solution above should fix this comment here as well, closing.